### PR TITLE
BFD-2726: Negative starting sequence number fix

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[BFD-2726](https://jira.cms.gov/browse/BFD-2726)

**User Story or Bug Summary:**

As a BFD engineer I want to ensure that the RDA pipeline jobs can only be executed with a zero or negative starting sequence number if running inside of the LoadRdaJsonApp.

---

### What Does This PR Do?

* Adds `map` call in `AppConfiguration` when loading starting sequence number values that force the value to be at least 1 when it's present.
* Refactors `AppConfiguration` to make it possible to test loading of different types of RDA job config separately.
* Adds unit tests for the RDA config loading methods.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by billie-rose — APPROVED** [review]

**Review by lsmitchell — COMMENTED** [review]
Thanks for the detailed testing! Just one thing I wasn't clear on.

**Review by daryll — APPROVED** [review]
lgtm

**Review by JeremyIR — DISMISSED** [review]

**Review by mackec1 — DISMISSED** [review]

File: apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java [hunk 1]
```diff
@@ -584,24 +582,48 @@ static RdaLoadOptions loadRdaLoadOptions(ConfigLoader config, IdHasher.Config id
...
+  /**
+   * Loads configuration settings used by {@link RdaServerJob}.
+   *
+   * @param config used to load configuration values
+   * @param grpcConfig settings for communicating with RDA API
+   * @param mockServerConfig used to construct the config settings
+   * @return a valid RdaServerJob.Config
+   */
+  @VisibleForTesting
+  static RdaServerJob.Config loadRdaServerJobConfig(
```
[the remark this answers is not in the document]
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java:623]: The `RdsServerJob` runs an RDA API server which is a mock RDA API server even though it's not a Mockito mock.  It's actually a full fledged RDA API implementation that can return random or synthetic data.  The original variable name reflected that it was the configuration for the mock server.

Of course in the unit test we do actually call this method with a Mockito mock of the `RdaServerJob.Config.ConfigBuilder`.

I can see that the old name is unclear given that we generally use the word mock in the context of Mockito and unit tests.  I'll rename the parameter to make it more obvious that it's not meant to be a Mockito mock but an actual server.

**Review by mjburling — APPROVED** [review]

**Review by aadedejifearless — APPROVED** [review]
Looks Good.

File: apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java:621]: Thanks!

**Pull request merged**